core-web/libs/portlets/edit-ema/portlet/src/lib/store/features/editor/toolbar/withUVEToolbar.ts

```diff
@@ -0,0 +1,131 @@
+                    languageId: Number(params?.language_id),
+                    pageURI: url,
+                    siteId: pageAPIResponse?.site?.identifier
+                });
+
+                const isPageLocked = computePageIsLocked(
+                    pageAPIResponse?.page,
+                    store.currentUser()
+                );
+                const shouldShowUnlock = isPageLocked && pageAPIResponse?.page.canLock;
+                const isExperimentRunning = experiment?.status === DotExperimentStatus.RUNNING;
+
+                const unlockButton = {
+                    inode: pageAPIResponse?.page.inode,
+                    loading: store.status() === UVE_STATUS.LOADING
+                };
+
+                const shouldShowInfoDisplay =
+                    !getIsDefaultVariant(pageAPIResponse?.viewAs.variantId) ||
+                    !store.canEditPage() ||
+                    isPageLocked ||
+                    !!store.device() ||
+                    !!store.socialMedia();
+
+                const siteId = pageAPIResponse?.site?.identifier;
+                const clientHost = `${params?.clientHost ?? window.location.origin}`;
+
+                return {
+                    editor: store.isPreviewModeActive()
+                        ? null
+                        : {
+                              bookmarksUrl,
+                              copyUrl: createFullURL(params, siteId),
+                              apiUrl: pageAPI
+                          },
+                    preview: store.isPreviewModeActive()
+                        ? {
+                              deviceSelector: {
+                                  apiLink: `${clientHost}${pageAPI}`,
+                                  hideSocialMedia: !store.isTraditionalPage()
+                              }
+                          }
+                        : null,
+                    currentLanguage: pageAPIResponse?.viewAs.language,
+                    urlContentMap: store.isEditState()
+                        ? (pageAPIResponse?.urlContentMap ?? null)
+                        : null,
+                    runningExperiment: isExperimentRunning ? experiment : null,
+                    workflowActionsInode: store.canEditPage() ? pageAPIResponse?.page.inode : null,
+                    personaSelector: {
+                        pageId: pageAPIResponse?.page.identifier,
+                        value: pageAPIResponse?.viewAs.persona ?? DEFAULT_PERSONA
+                    },
+                    unlockButton: shouldShowUnlock ? unlockButton : null,
+                    showInfoDisplay: shouldShowInfoDisplay
+                };
+            })
+        })),
+        withMethods((store) => ({
+            // Fake method to toggle preview mode
+            // This method should be implemented in the real application
+            togglePreviewMode: (preview) => {
+                patchState(store, {
+                    isPreviewModeActive: preview
```

Comment:
Are we sure we want a boolean here?

Reply:
Yes, this represent if the user is in the preview mode (whether the user pressed the eye-button or entered it manually with queryParams)

Reply:
We might have a loading state when we move from one toolbar to the other...

Reply:
Oh i see the point, i will update the task on Preview Mode button to add a `state` prop here